src/user/user.repository.ts

```diff
@@ -1,16 +1,15 @@
-import { Connection, EntityRepository, Repository } from 'typeorm';
+import { Connection, EntityRepository, Repository } from "typeorm"
+import { User } from "./user.entity"
 
 @EntityRepository(User)
 export class UserRepository extends Repository<User> {
-    async findAll(): Promise<[User[], number]> {
-        return await this
-            .createQueryBuilder('user')
-            .getManyAndCount();
-    }
+  async findAll(): Promise<[User[], number]> {
+    return await this.createQueryBuilder("user").getManyAndCount()
+  }
 }
 
-export const UserRepositoryProvider = {
-    provide: 'UserRepository',
-    useFactory: (connection: Connection) => connection.getCustomRepository(UserRepository),
-    inject: [Connection]
-};
+// export const UserRepositoryProvider = {
```

Comment:
Remove this.

This workaround was needed back in Nest 4 for custom repositories.